modules/apps/layout/layout-service/src/main/java/com/liferay/layout/internal/upgrade/v1_4_3/LayoutClassedModelUsageUpgradeProcess.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.layout.internal.upgrade.v1_4_3;
+
+import com.liferay.document.library.kernel.model.DLFileEntry;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.LoggingTimer;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Gergely Szalay
+ */
+public class LayoutClassedModelUsageUpgradeProcess extends UpgradeProcess {
+
+	public LayoutClassedModelUsageUpgradeProcess(
+		ClassNameLocalService classNameLocalService) {
+
+		_dlFileEntryClassNameId = classNameLocalService.getClassNameId(
+			DLFileEntry.class.getName());
+		_fileEntryClassNameId = classNameLocalService.getClassNameId(
+			FileEntry.class.getName());
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (LoggingTimer loggingTimer = new LoggingTimer();
+			Statement statement = connection.createStatement();
+			ResultSet resultSet = statement.executeQuery(
+				"select layoutClassedModelUsageId from " +
```

Comment:
Hi @gergelyszalay, there's no need to do a select and then iterate, you can do it with one update, that will speed things up.